lib/path-utils.ts

```diff
@@ -14,24 +14,61 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { normalize, relative } from 'path';
+import * as _ from 'lodash';
+import * as path from 'path';
+import { posix } from 'path';
 
 // Export all of node's path functions, so that users of this
 // module need only import this file
 export * from 'path';
 
+const nativeSepRE = new RegExp(_.escapeRegExp(path.sep), 'g');
+const nativeDotDotRE = new RegExp('^\\.\\.$|\\.\\.' + _.escapeRegExp(path.sep));
+
 /**
- * Given two paths, check whether the first contains the second
- * @param path1 The potentially containing path
- * @param path2 The potentially contained path
+ * Given two native (platform-dependent) paths, check whether the first
+ * contains the second
+ * @param path1 The potentially containing native path
+ * @param path2 The potentially contained native path
  * @return A boolean indicating whether `path1` contains `path2`
  */
 export const contains = (path1: string, path2: string): boolean => {
 	// First normalise the input, to remove any path weirdness
-	path1 = normalize(path1);
-	path2 = normalize(path2);
+	path1 = path.normalize(path1);
+	path2 = path.normalize(path2);
+
+	// Now test if any part of the relative path contains a .. ,
+	// which would tell us that path1 is not part of path2
+	return !nativeDotDotRE.test(path.relative(path1, path2));
```

Comment:
While this `resin-multibuild` PR is blocked by the [`resin-bundle-resolve`](https://github.com/balena-io-modules/resin-bundle-resolve/pull/36) one... I was working on the CLI and I came across the need for the `path-utils` `contains()` function in native (non-POSIX) flavour. The previous commit in this PR had modified the `contains` function to use POSIX paths (because it is part of the fix for Windows path handling when using the CLI on Windows), but I am now thinking that it would be good to have both native and POSIX versions of `contains` available for use. There are several possibilities on how to refactor `path-utils.ts` and I am not clear that one is much better than the others, and what I've come up with so far was to keep the original `contains` with the same name, and add `posixContains`. Something I did not want to do was to add "if tests" in the body of the `contains` function, because this would likely slow down processing of tar streams.